pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -271,7 +283,13 @@ func (r *repository) Get(ctx context.Context, dgst digest.Digest, options ...dis
 	}
 
 	ref := imageapi.DockerImageReference{Namespace: r.namespace, Name: r.name, Registry: r.registryAddr}
-	manifest, err := r.manifestFromImageWithCachedLayers(image, ref.DockerClientDefaults().Exact())
+	if managed := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; managed == "true" {
+		ref.Registry = ""
```

Comment:
what does an empty registry here do? don't we still need to default tag and namespace?